src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -535,17 +564,17 @@ private LocalDefinition EmitReceiverRef(BoundExpression receiver, bool isAccessC
             }
 
             Debug.Assert(receiverType.IsVerifierValue());
-            return EmitAddress(receiver, AddressKind.Writeable);
+            return EmitAddress(receiver, addressKind);
         }
 
         /// <summary>
         /// May introduce a temp which it will return. (otherwise returns null)
         /// </summary>
-        private LocalDefinition EmitInstanceFieldAddress(BoundFieldAccess fieldAccess)
+        private LocalDefinition EmitInstanceFieldAddress(BoundFieldAccess fieldAccess, bool isReadonly)
```

Comment:
Constrained allows mutating method calls, but does not allow taking field addresses, so  it needs to become Writeable, when digging through fields. I had AddressKind propagated initially, but it was prone to errors where receiver address was taken with Constrained where it should be Writable.

With a bool  "writeable" it is harder to make such mistake.

---
In reply to: [108265271](https://github.com/dotnet/roslyn/pull/18175#discussion_r108265271) [](ancestors = 108265271)